lib/unparser/emitter/literal/primitive.rb

```diff
@@ -7,10 +7,41 @@ class Primitive < self
 
         children :value
 
+        # Emitter for string literals
+        class String < self
+          handle :str
+
+        private
+
+          # Dispatch value
+          #
+          # @return [undefined]
+          #
+          # @api private
+          #
+          def dispatch
+            if value.include? "'"
+              # Write %()-style strings for strings with both double and single
+              # quotes
+              if value.include? '"'
+                write("%(#{value})")
```

Comment:
@glittershark I think we simply should go in a loop trying each one. When all are exhausted we simply fall back to inspect. That way adding other reasonable alternatives like `%[]` is very easy.